### PR TITLE
Created DRA shell script to create artifacts and dependency report

### PR DESCRIPTION
**Pull request by andsel**
## Release notes

[rn:skip] 

## What does this PR do?

Introduces a bash script to build all the artifacts and dependencies report.

## Checklist

- [x] My code follows the style guidelines of this project
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~ ] I have made corresponding change to the default configuration files (and/or docker env variables)~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~

## Author's Checklist

- [x] run the artifact creation script with `ci/dra.sh` and check all artifacts and dependencies report is created

## How to test this PR locally
run the artifact creation script with 
```
ci/dra.sh
``` 
and check all artifacts and dependencies report is created

## Related issues

-

### CODE REVIEW

File: ci/dra.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# install rake's gme
+gem install rake
+
+if [ -z "$VERSION_QUALIFIER_OPT" ]; then
+  RELEASE=1 rake artifact:all
+else
+  VERSION_QUALIFIER="$VERSION_QUALIFIER_OPT" RELEASE=1 rake artifact:all
+fi
+
+STACK_VERSION=`cat versions.yml | sed -n 's/^logstash\:\s\([[:digit:]]*\.[[:digit:]]*\.[[:digit:]]*\)$/\1/p'`
```
Comment (jsvd) [inline ci/dra.sh:18]: Is this injected from the environment by any chance? it could spare us from having to do this trick

File: ci/dra.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash -ie
+#Note - ensure that the -e flag is set to properly set the $? status if any command fails
+
+# Since we are using the system jruby, we need to make sure our jvm process
+# uses at least 1g of memory, If we don't do this we can get OOM issues when
+# installing gems. See https://github.com/elastic/logstash/issues/5179
+export JRUBY_OPTS="-J-Xmx1g"
+
+# install rake's gme
```
Comment (jsvd) [inline ci/dra.sh:9]: unnecessary comment:

```suggestion
```

JRuby already comes with rake, do we really need to install it?

```
/tmp/jruby-9.3.7.0
❯ find . -name "*rake*"
./bin/rake.bat
./bin/rake
./lib/ruby/stdlib/rake
./lib/ruby/stdlib/rake/ant/tasks/raketasks.rb
./lib/ruby/stdlib/rake/ant/rake.rb
./lib/ruby/stdlib/rubygems/ext/rake_builder.rb
./lib/ruby/stdlib/bundler/vendor/thor/lib/thor/rake_compat.rb
./lib/ruby/gems/shared/specifications/default/rake-ant-1.0.4.gemspec
./lib/ruby/gems/shared/specifications/rake-12.3.3.gemspec
./lib/ruby/gems/shared/cache/rake-12.3.3.gem
./lib/ruby/gems/shared/gems/rake-12.3.3
./lib/ruby/gems/shared/gems/rake-12.3.3/bin/rake
./lib/ruby/gems/shared/gems/rake-12.3.3/lib/rake
./lib/ruby/gems/shared/gems/rake-12.3.3/lib/rake/rake_module.rb
./lib/ruby/gems/shared/gems/rake-12.3.3/lib/rake/rake_test_loader.rb
./lib/ruby/gems/shared/gems/rake-12.3.3/lib/rake.rb
./lib/ruby/gems/shared/gems/rake-12.3.3/doc/proto_rake.rdoc
./lib/ruby/gems/shared/gems/rake-12.3.3/doc/rake.1
./lib/ruby/gems/shared/gems/rake-12.3.3/doc/rakefile.rdoc
./lib/ruby/gems/shared/gems/rake-12.3.3/exe/rake
./lib/ruby/gems/shared/gems/rake-12.3.3/rake.gemspec
```

Comment (andsel) [reply ci/dra.sh:9]: So `rake` it's already bundled with JRuby?

Comment (jsvd) [reply ci/dra.sh:9]: yeah it comes with a few bundled gems: 

```
/tmp/jruby-9.3.7.0
❯ ls ./lib/ruby/gems/shared/gems
bundler-2.2.29		jar-dependencies-0.4.1	power_assert-1.1.3	rdoc-6.1.2.1		xmlrpc-0.3.0
did_you_mean-1.3.0	minitest-5.11.3		racc-1.5.2-java		rubygems-update-3.2.29
irb-1.0.0		net-telnet-0.1.1	rake-12.3.3		test-unit-3.2.9

```

File: ci/dra.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  RELEASE=1 rake artifact:all
+else
+  VERSION_QUALIFIER="$VERSION_QUALIFIER_OPT" RELEASE=1 rake artifact:all
+fi
+
+STACK_VERSION=`cat versions.yml | sed -n 's/^logstash\:\s\([[:digit:]]*\.[[:digit:]]*\.[[:digit:]]*\)$/\1/p'`
+
+echo "Creating dependencies report for ${STACK_VERSION}"
+mkdir -p build/reports/dependencies-reports/
+bin/dependencies-report --csv=build/reports/dependencies-reports/logstash-${STACK_VERSION}.csv
```
Comment (jsvd) [inline ci/dra.sh:19]: I suggest at least printing a final success message, with some information about what was generated, for example:


```
❯ for file in build/logstash-*; do shasum $file;done  
63a4578a28bb88f461494cc1619eb6bb2dcb385a  build/logstash-8.5.0-aarch64.rpm
c019dbe419dd98cd04242da6d725488a55195ae9  build/logstash-8.5.0-amd64.deb
a183a0aa0343a06392c9a8914efa1a7a4b1a2f5a  build/logstash-8.5.0-arm64.deb
3ab631d1236bd34b5657a154727e6737c08d2209  build/logstash-8.5.0-darwin-aarch64.tar.gz
d8fb95b1de46c7246315d7bce183d1dba9d7264a  build/logstash-8.5.0-darwin-x86_64.tar.gz
c6775dfdaafdecc009ba7a9754d53f37539a9e09  build/logstash-8.5.0-linux-aarch64.tar.gz
e9d38879f76db4e314bec31e79bcd9c6aa92d810  build/logstash-8.5.0-linux-x86_64.tar.gz
c03f84a1901d5980464a4b25cd8588f96322b9d5  build/logstash-8.5.0-no-jdk.deb
d5f176e24ac60c2d51209a9a2107c5ce71c0ea8c  build/logstash-8.5.0-no-jdk.rpm
d4d4c572f16591255ef6ec7698e45253f227c340  build/logstash-8.5.0-no-jdk.tar.gz
b8793b727ffda82a9556e9d4d2ae94f735a6ea85  build/logstash-8.5.0-no-jdk.zip
4985c4e58baec53d1c184652002d5a8f420fa5fa  build/logstash-8.5.0-windows-x86_64.zip
36f18c9de1c61230ba0a36779c8da98a317dcca0  build/logstash-8.5.0-x86_64.rpm
c50b7232ad553571c90ce5185c8515372df07319  build/logstash-ironbank-8.5.0-docker-build-context.tar.gz
35a66975ef01087ff31a680ee8ae0ed8485d2d65  build/logstash-oss-8.5.0-aarch64.rpm
b9fc21c0efcccc0f53a19de1cfaefc9a6b4eccb5  build/logstash-oss-8.5.0-amd64.deb
85dbc0e37824e7ea90373429c2a981f0c0804222  build/logstash-oss-8.5.0-arm64.deb
35b88312fcf1410cec8bea0433fdbf4ef2a8854e  build/logstash-oss-8.5.0-darwin-aarch64.tar.gz
fc31ab1ff6c48ad31af09d261d60bb44c0bac84d  build/logstash-oss-8.5.0-darwin-x86_64.tar.gz
1d7f5dacd1388214cd28f539d1c6d301a37ac76b  build/logstash-oss-8.5.0-linux-aarch64.tar.gz
3c55047b5d32ca581e4adc4fd22ce69c53b0c4df  build/logstash-oss-8.5.0-linux-x86_64.tar.gz
5a0fc122b1ae7f6a5e3c180f4947940172778d7a  build/logstash-oss-8.5.0-no-jdk.deb
0c7583fdad688e96d076ad78035f343c4c2f63ab  build/logstash-oss-8.5.0-no-jdk.rpm
6e17613ef26c1399471869a8f8ccadd5c1368581  build/logstash-oss-8.5.0-no-jdk.tar.gz
5f17c0bc4a58c9cd88797f134788d8151607d6a1  build/logstash-oss-8.5.0-no-jdk.zip
1d195e45ce871ca3363d3eb3144ad89c76807cec  build/logstash-oss-8.5.0-windows-x86_64.zip
abc6a0b8c008e404ede34b26503f675ff12818e4  build/logstash-oss-8.5.0-x86_64.rpm
```

**Review by jsvd — COMMENTED** [review]
Added a suggestion to improve messaging when the build completes.
Also, it seems that parts of the build can fail, but the overall build still succeeds.
It'd be nice to solve this to avoid surprises, and/or having to double check each build .

**Comment by andsel** [comment]
@jsvd fixed the issue adding `set -e` in the outer shell script.

**Review by jsvd — APPROVED** [review]
LGTM

**Pull request merged**